src/NHibernate/Logging.cs

```diff
@@ -349,39 +360,174 @@ public bool IsEnabled(InternalLogLevel logLevel)
+		public void Info(object message) => _this.Info(message?.ToString());
+
+		// Since v5.1
+		[Obsolete("Please use Info(string, params object[]) INHibernateLogger extension method instead.")]
+		public void Info(string message) => _this.Info(message, default(object[]));
+
+		// Since v5.1
+		[Obsolete("Please use Info(Exception, string, params object[]) INHibernateLogger extension method instead.")]
+		public void Info(object message, Exception exception) => _this.Info(exception, message?.ToString());
+
+		// Since v5.1
+		[Obsolete("Please use Info(string, params object[]) INHibernateLogger extension method instead.")]
+		public void InfoFormat(string format, params object[] args) => _this.Info(format, args);
+
+		// Since v5.1
+		[Obsolete("Please use Warn(string, params object[]) INHibernateLogger extension method instead.")]
+		public void Warn(object message) => _this.Warn(message?.ToString());
+
+		// Since v5.1
+		[Obsolete("Please use Warn(string, params object[]) INHibernateLogger extension method instead.")]
+		public void Warn(string message) => _this.Warn(message, default(object[]));
+
+		// Since v5.1
+		[Obsolete("Please use Warn(Exception, string, params object[]) INHibernateLogger extension method instead.")]
+		public void Warn(object message, Exception exception) => _this.Warn(exception, message?.ToString());
+
+		// Since v5.1
+		[Obsolete("Please use Warn(string, params object[]) INHibernateLogger extension method instead.")]
+		public void WarnFormat(string format, params object[] args) => _this.Warn(format, args);
+	}
+
+	/// <summary>
+	/// Factory building loggers which log nothing.
+	/// </summary>
+#pragma warning disable 618 // ILoggerFactory is obsolete
+	public class NoLoggingLoggerFactory: INHibernateLoggerFactory, ILoggerFactory
+#pragma warning restore 618
 	{
 		private static readonly INHibernateLogger Nologging = new NoLoggingInternalLogger();
-		public INHibernateLogger LoggerFor(string keyName)
+		/// <inheritdoc />
+		INHibernateLogger INHibernateLoggerFactory.LoggerFor(string keyName)
 		{
 			return Nologging;
 		}
 
-		public INHibernateLogger LoggerFor(System.Type type)
+		/// <inheritdoc />
+		INHibernateLogger INHibernateLoggerFactory.LoggerFor(System.Type type)
 		{
 			return Nologging;
 		}
+
+#pragma warning disable 618
+		[Obsolete("Use this as an INHibernateLoggerFactory instead.")]
+		public IInternalLogger LoggerFor(System.Type type)
+		{
+			return Nologging;
+		}
+
+		[Obsolete("Use this as an INHibernateLoggerFactory instead.")]
+		public IInternalLogger LoggerFor(string keyName)
+		{
+			return Nologging;
+		}
+#pragma warning restore 618
 	}
 
-	public class NoLoggingInternalLogger: INHibernateLogger
+#pragma warning disable 618 // NHibernateLoggerBase is obsolete, to be removed in a upcoming major version
+	/// <summary>
+	/// Logger which logs nothing.
+	/// </summary>
+	public class NoLoggingInternalLogger: NHibernateLoggerBase, INHibernateLogger
```

Comment:
Adding a base class or changing an interface for a more derived one is not a breaking change.